### PR TITLE
fix(suite): move types to components and delete unused ones

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Part of #9616

### CODE REVIEW

File: packages/components/src/components/form/inputTypes.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type InputState = 'success' | 'warning' | 'error';
+
+export type InputVariant = 'small' | 'large';
```
Comment (jvaclavik) [inline packages/components/src/components/form/inputTypes.ts]: these two types are related to multiple form components, so I created a new shared file only for form components.

File: packages/components/src/components/typography/Link/Link.tsx [hunk 2]
```diff
@@ -3,7 +3,8 @@ import styled, { css } from 'styled-components';
 import { Icon, IconProps } from '../../assets/Icon/Icon';
 import { useTheme } from '../../../utils';
 import { FONT_SIZE } from '../../../config/variables';
-import { ParagraphSize } from '../../../support/types';
+
+export type LinkSize = 'normal' | 'small' | 'tiny';
```
Comment (jvaclavik) [inline packages/components/src/components/typography/Link/Link.tsx:6]: This type is the same as ParagraphSize, but I think we don't need to share these. WDYT?

File: packages/components/src/components/Box/Box.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { ReactNode, HTMLAttributes } from 'react';
 import styled, { css } from 'styled-components';
-import { InputState } from '../../support/types';
 import { getInputStateBorderColor } from '../form/InputStyles';
+import { InputState } from '../form/inputTypes';
```
Comment (jvaclavik) [inline packages/components/src/components/Box/Box.tsx]: Not sure about this one. In this case, it's maybe not just an input type. Should I make it more generic (but we are going to create another more global type) or duplicate InputState for this component? What do you think @dahaca?

Comment (dahaca) [reply packages/components/src/components/Box/Box.tsx]: Honestly I am sure I removed it in the `Input` PR.. Maybe I did not push. Anyways, `Box` should not be using `InputState` of course, also `InputState` does not need the `success` state anymore, I believe.

Comment (jvaclavik) [reply packages/components/src/components/Box/Box.tsx]: I created separate type for Box for now

File: packages/components/src/components/typography/Link/Link.tsx (hunk 2 above)
Comment (komret) [reply packages/components/src/components/typography/Link/Link.tsx:6]: This is going to be reworked as part of https://github.com/trezor/trezor-suite/issues/9618, we can have it duplicate for now.

File: packages/components/src/components/form/inputTypes.ts (hunk 1 above)
Comment (komret) [reply packages/components/src/components/form/inputTypes.ts]: Sounds good.

**Review by komret — APPROVED** [review]

**Comment by komret** [comment]
I resolved a conflict.

**Pull request merged**